pkg/controller/garbagecollector/graph_builder.go

```diff
@@ -118,36 +121,45 @@ func listWatcher(client *dynamic.Client, resource schema.GroupVersionResource) *
 }
 
 func (gb *GraphBuilder) controllerFor(resource schema.GroupVersionResource, kind schema.GroupVersionKind) (cache.Controller, error) {
-	// TODO: consider store in one storage.
-	glog.V(5).Infof("create storage for resource %s", resource)
-	client, err := gb.metaOnlyClientPool.ClientForGroupVersionKind(kind)
-	if err != nil {
-		return nil, err
-	}
-	gb.registeredRateLimiterForControllers.registerIfNotPresent(resource.GroupVersion(), client, "garbage_collector_monitoring")
-	setObjectTypeMeta := func(obj interface{}) {
+	setObjectTypeMeta := func(obj interface{}, clone bool) (runtime.Object, error) {
 		runtimeObject, ok := obj.(runtime.Object)
 		if !ok {
-			utilruntime.HandleError(fmt.Errorf("expected runtime.Object, got %#v", obj))
+			return nil, fmt.Errorf("expected runtime.Object, got %#v", obj)
+		}
+		if clone {
+			copy, err := api.Scheme.DeepCopy(runtimeObject)
```

Comment:
Or we could not clone and instead include the GVK as part of the event object (i.e. use a custom struct that has a runtime.Object and GVK).